src/corefx/System.Globalization.Native/pal_collation.c

```diff
@@ -143,7 +143,7 @@ static int AddItem(UCharList* list, const UChar item)
     if (size >= list->capacity)
     {
         list->capacity *= 2;
```

Comment:
Note: At present this cannot overflow. It is called only from GetCustomRules and given the current set of rules the resize can happen only one time (from 512 elements to 1024, of which only around 860 are used in the most expensive code path). It may be worth adding an `assert` to verify that the overflow doesn't happen here.